### PR TITLE
[all] Update containers to support c++x11 features

### PR DESCRIPTION
**Pull request by matthieu-nesme**
solving #107 

Adding move constructor and move assignment operator to helper containers.

### CODE REVIEW

File: SofaKernel/framework/sofa/helper/set.h [hunk 1]
```diff
@@ -2,18 +2,18 @@
 *       SOFA, Simulation Open-Framework Architecture, development version     *
 *                (c) 2006-2016 INRIA, USTL, UJF, CNRS, MGH                    *
 *                                                                             *
-* This library is free software; you can redistribute it and/or modify it     *
+* _set library is free software; you can redistribute it and/or modify it     *
```
Comment (guparan) [inline SofaKernel/framework/sofa/helper/set.h]: I guess some "find+replace" went wrong here.

**Comment by matthieu-nesme** [comment]
:D corrected

File: SofaKernel/framework/sofa/helper/set.h (hunk 1 above)
Comment (guparan) [reply SofaKernel/framework/sofa/helper/set.h]: You forgot this one ;)

File: SofaKernel/framework/sofa/helper/set.h [hunk 2]
```diff
@@ -13,7 +13,7 @@
 * for more details.                                                           *
 *                                                                             *
 * You should have received a copy of the GNU Lesser General Public License    *
-* along with this library; if not, write to the Free Software Foundation,     *
+* along with _set library; if not, write to the Free Software Foundation,     *
```
Comment (guparan) [inline SofaKernel/framework/sofa/helper/set.h]: And this one ;)

**Review by guparan — CHANGES_REQUESTED** [review]
Thanks for your fix but you forgot two "_set" typos :(

**Comment by matthieu-nesme** [comment]
Corrected again...
This code is pure c++, no need for special assignees.
I let you update your changelog if you want to incorporate this PR.

**Comment by damienmarchal** [comment]
@guparan  Is there is still something to change ? 
As this does not change the API or the scene it can be merged for the 16.12

**Comment by JeremieA** [comment]
Hello,

The issue I have with this approach is that this make it very invasive, as Sofa is adding operators in std namespace for all types, even not its own. So within an application that could also does the same, we would get conflicts / redefinition / ambiguity compiler errors, or worse silently override the application's own operators that could have relied on a different string syntax.

I agree that the approach of subclassing the standard containers just to be able to package them within a Data is wrong. In our current branch we are refactoring the approach, relying more on `DataTypeInfo` to generically access features like string parsing, without having to put them in the type itself. But this is work in progress so it is not ready to be pushed yet. So in the meantime this change could be pushed, but with the understanding that it may be a temporary solution only.

Jeremie.

**Comment by matthieu-nesme** [comment]
@JeremieA I was waiting for this remark :) I totally agree. My point was to wait for an eventual (quite improbable) conflict. But if you think the conflict could be silent, indeed it's a problem.

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
This PR: 
  - [ ] successfully compile on each configuration on the CI.
  - [x] does not generates new warnings or test failure. 
  - [x] does not break existing scenes.
  - [x] does not break API compatibility.  
  - [x] is now 1 week old.
  - [x] has been reviewed and agreed to be transitional.

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]
Tada !!

**Comment by guparan** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]
Tudu !

**Comment by guparan** [comment]
[ci-build]

**Comment by guparan** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
Hi all,

I have opened a new PR in WIP to handle the cleaning of set & vector.

It currently contains an attempt to remove the std namespace "pollution" in commit https://github.com/sofa-framework/sofa/pull/278/commits/d3ecab1eaa7f2f29242aacf646910b2eb3ab47dd

Feedback & contribution in #278

**Pull request merged**